engine/schema/src/com/cloud/network/dao/PhysicalNetworkServiceProviderVO.java

```diff
@@ -308,15 +331,21 @@ public void setEnabledServices(List<Service> services) {
         if (this.isSecuritygroupServiceProvided()) {
             services.add(Service.SecurityGroup);
         }
+        if (this.isDynamicRoutingProvided()) {
+            services.add(Service.VPCDynamicRouting);
+        }
         return services;
     }
 
     @Override
-    public boolean isNetworkAclServiceProvided() {
-        return networkAclServiceProvided;
+    public String toString() {
+        return "PhysicalNetworkServiceProviderVO [id=" + id + ", physicalNetworkId=" + physicalNetworkId + ", destPhysicalNetworkId=" + destPhysicalNetworkId + ", providerName="
+                + providerName + ", state=" + state + ", vpnServiceProvided=" + vpnServiceProvided + ", dhcpServiceProvided=" + dhcpServiceProvided + ", dnsServiceProvided="
+                + dnsServiceProvided + ", gatewayServiceProvided=" + gatewayServiceProvided + ", firewallServiceProvided=" + firewallServiceProvided + ", sourcenatServiceProvided="
+                + sourcenatServiceProvided + ", lbServiceProvided=" + lbServiceProvided + ", staticnatServiceProvided=" + staticnatServiceProvided
+                + ", portForwardingServiceProvided=" + portForwardingServiceProvided + ", userdataServiceProvided=" + userdataServiceProvided + ", securitygroupServiceProvided="
+                + securitygroupServiceProvided + ", networkAclServiceProvided=" + networkAclServiceProvided + ", dynamicRoutingProvided=" + dynamicRoutingProvided + ", removed="
+                + removed + "]";
```

Comment:
Given number of concatenations, please refactor to use a ``StringBuilder`` in order to reduce the number of strings created.